### ISSUE TITLE
Gutenberg: Featured Image not updated when edited on Media Modal

### ISSUE THREAD

**Issue by mmtr** [Automattic/wp-calypso#29829]
#### Steps to reproduce
1. Open the block editor at /block-editor and select a site under a paid plan (free plans don't allow to edit images in the Media Modal).
2. Set a featured image.
3. Click on the featured media so the Medial Modal is opened.
4. Select the current featured image and click on the Edit button.
5. Push the "Edit Image" button and apply some changes to the image (e.g. flip it).
6. Click on the Done button and insert the modified image.

#### What I expected
The featured image should display the modified image.

#### What happened instead
The featured image displays the previous image.

#### Context / Source
Related: #29791.

**Comment by obenland** [comment]
I was able to reproduce that. It appears it doesn't update the view of that picture, it displays correctly after a reload

**Comment by Copons** [comment]
Calypso Media Modal successfully updates the media URL (therefore it shows correct once reloaded), but there is no handling in the iframe bridge.

Unfortunately it's easier said than done, as the post object in the Block Editor uses only the media ID for the `featured_media` attribute, which does not change when the media is edited.

I'm trying to figure out if there is a way to force the featured image component to reselect the media with the new URL. 🤔

**Comment by mmtr** [comment]
In #29820, we solved a similar problem with Simple Payments. We refactored the logic in order to use multiple attributes (ID, URL, etc) as source of truth of the product images instead of just the ID, so we can specifically change the URL of the image.

That approach is a bit harder to replicate here since it will require a change in core.

Another option we can try is to invalidate somehow the [media object stored on the state](https://github.com/WordPress/gutenberg/blob/39d435bc964cd119feea3f763450d0e8436ce59e/packages/editor/src/components/post-featured-image/index.js#L108), so the `PostFeaturedImage` component will retrieve again the data from the API if we toggle quickly the ID:

```js
editPost( { featured_media: 0 } ); 
// Invalidate here the cached featured image so next call 
// will retrieve most recent data from API.
editPost( { featured_media: image.id } );
```

Maybe this [`shouldInvalidate`](https://github.com/WordPress/gutenberg/blob/39d435bc964cd119feea3f763450d0e8436ce59e/packages/core-data/src/resolvers.js#L71-L78) function can help.
